x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -47,6 +66,18 @@
 public class AggregateDoubleMetricFieldMapper extends FieldMapper {
 
     public static final String CONTENT_TYPE = "aggregate_metric_double";
+    public static final String SUBFIELD_SEPARATOR = ".";
```

Comment:
When I changed the `SUBFIELD_SEPARATOR` from `._` to `.` the yaml test `Test term query on aggregate metric field` started failing because it returned 0 hits. When I change the separator back to `._`, test becomes successful.

I debugged the test but values seem to be stored correctly. @polyfractal @jimczi any idea what I am missing here?

Link to test: https://github.com/elastic/elasticsearch/blob/b9bf2599eec5896d0c2412373231cb3700bc4442/x-pack/plugin/src/test/resources/rest-api-spec/test/aggregate-metrics/10_basic.yml#L57

Reply:
Issue solved. Many thanks to @polyfractal for spotting the problem.